ReactCommon/react/renderer/uimanager/UIManagerBinding.cpp

```diff
@@ -577,48 +635,59 @@ jsi::Value UIManagerBinding::get(
+          auto nativeMeasurerValue =
+              runtime.global()
+                  .getProperty(runtime, "__turboModuleProxy")
+                  .asObject(runtime)
+                  .asFunction(runtime)
+                  .call(runtime, "NativeFabricMeasurerTurboModule");
+          auto onSuccessFunction =
+              arguments[1].getObject(runtime).getFunction(runtime);
 
           if (nativeMeasurerValue.isObject()) {
-              // This calls measureNatively if the NativeFabricMeasurerTurboModule is found.
-              // The return value doesn't matter here because the measure values will be passed through the callback.
-              jsi::Value returnValue = nativeMeasurerValue.asObject(runtime).getPropertyAsFunction(runtime, "measureNatively")
-                      .call(runtime, shadowNode.get()->getTag(), arguments[1].getObject(runtime).getFunction(runtime));
-              turboModuleCalled = true;
+            // This calls measureNatively if the NativeFabricMeasurerTurboModule
+            // is found. The return value doesn't matter here because the
+            // measure values will be passed through the callback.
+            jsi::Value returnValue =
+                nativeMeasurerValue.asObject(runtime)
+                    .getPropertyAsFunction(runtime, "measureNatively")
+                    .call(
+                        runtime,
+                        shadowNode.get()->getTag(),
+                        // measureNatively takes two callbacks, one for if the
+                        // layout is found and one for if an error occurs so we
+                        // can gracefully fallback to measuring out of the
+                        // shadow tree.
+                        onSuccessFunction,
+                        jsi::Function::createFromHostFunction(
+                            runtime,
+                            jsi::PropNameID::forAscii(
+                                runtime, "onNativeMeasureFail"),
+                            1,
+                            [uiManager, shadowNode](
+                                jsi::Runtime &rt,
+                                const jsi::Value &thisVal,
+                                const jsi::Value *args,
+                                size_t count) {
+                              // We return onSuccessFunction out of the onFail
+                              // instead of using the one defined above to avoid
+                              // a scoping bug.
+                              auto onSuccessFunction =
+                                  args[0].getObject(rt).getFunction(rt);
+                              measureFromShadowTree(
+                                  uiManager, shadowNode, onSuccessFunction, rt);
+                              return jsi::Value::undefined();
+                            }));
+            turboModuleCalled = true;
           }
 
           if (turboModuleCalled) {
```

Comment:
we're setting turboModuleCalled = true two lines above this. why check now if it's set? why not just return Value::undefined() above?

Reply:
Probably was a rough edge around the initial implementation of this. Your recommendation sounds like a cleaner way to do it.